client/signup/steps/site-or-domain/index.jsx

```diff
@@ -120,8 +128,11 @@ class SiteOrDomain extends Component {
 	}
 
 	renderScreen() {
+		const productsLoaded = ! isEmpty( this.props.productsList );
+
 		return (
 			<div>
+				{ ! productsLoaded && <QueryProductsList /> }
```

Comment:
Yep, if it's already fetched it will be pulled from cache, and if it's the same the UI won't update so all cool. :)